### PR TITLE
[mlir][spirv] Improve coop matrix attribute handling

### PR DESCRIPTION
**Pull request by kuhar**
- Fix values of Matrix Operand bit enums.
- Add verification for the aligned Memory Operand attributes. Mark the 'Aligned' enumerant as not supported.

The target test passes validation with `spirv-val`.

### CODE REVIEW

**Comment by kuhar** [comment]
Seems like the failure is just buildkite having a bad day.

**Comment by kuhar** [comment]
(Rebased to retrigger CI)

File: mlir/include/mlir/Dialect/SPIRV/IR/SPIRVBase.td [hunk 1]
```diff
@@ -4077,11 +4077,11 @@ def SPIRV_KHR_CooperativeMatrixLayoutAttr :
...
-def SPIRV_KHR_CMO_MatrixA_Signed : I32BitEnumAttrCaseBit<"ASigned", 1>;
-def SPIRV_KHR_CMO_MatrixB_Signed : I32BitEnumAttrCaseBit<"BSigned", 2>;
-def SPIRV_KHR_CMO_MatrixC_Signed : I32BitEnumAttrCaseBit<"CSigned", 4>;
-def SPIRV_KHR_CMO_Result_Signed  : I32BitEnumAttrCaseBit<"ResultSigned", 8>;
-def SPIRV_KHR_CMO_AccSat         : I32BitEnumAttrCaseBit<"AccSat", 16>;
+def SPIRV_KHR_CMO_MatrixA_Signed : I32BitEnumAttrCaseBit<"ASigned", 0>;
+def SPIRV_KHR_CMO_MatrixB_Signed : I32BitEnumAttrCaseBit<"BSigned", 1>;
+def SPIRV_KHR_CMO_MatrixC_Signed : I32BitEnumAttrCaseBit<"CSigned", 2>;
+def SPIRV_KHR_CMO_Result_Signed  : I32BitEnumAttrCaseBit<"ResultSigned", 3>;
+def SPIRV_KHR_CMO_AccSat         : I32BitEnumAttrCaseBit<"AccSat", 4>;
```
Comment (qedawkins) [inline mlir/include/mlir/Dialect/SPIRV/IR/SPIRVBase.td:4084]: The prior enum values look to match the values given by the extension. Is there a reason to change it to `0` through `4` here? https://htmlpreview.github.io/?https://github.com/KhronosGroup/SPIRV-Registry/blob/master/extensions/KHR/SPV_KHR_cooperative_matrix.html

Comment (qedawkins) [reply mlir/include/mlir/Dialect/SPIRV/IR/SPIRVBase.td:4084]: resolved offline; BitEnum specifies the bit index rather than the value.

**Review by qedawkins — APPROVED** [review]

**Pull request merged**